### PR TITLE
BAEL-4384: Invoke a private method using reflection

### PR DESCRIPTION
**Pull request by bradhandy**
Added a test case to show the process for invoking a private method via the Java Reflection API.

### CODE REVIEW

File: core-java-modules/core-java-reflection-2/src/test/java/com/baeldung/reflection/access/privatemethods/InvokePrivateMethodsUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package com.baeldung.reflection.access.privatemethods;
+
+import com.google.common.primitives.Longs;
+import org.junit.jupiter.api.Test;
+
+import java.lang.reflect.Method;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class InvokePrivateMethodsUnitTest {
```
Comment (ashleyfrieze) [inline core-java-modules/core-java-reflection-2/src/test/java/com/baeldung/reflection/access/privatemethods/InvokePrivateMethodsUnitTest.java:10]: JUnit 5 doesn't need the class to be `public`

File: core-java-modules/core-java-reflection-2/src/test/java/com/baeldung/reflection/access/privatemethods/InvokePrivateMethodsUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+    private final long[] someLongArray = new long[] { 1L, 2L, 3L, 4L, 5L };
+
+    @Test
+    public void whenSearchingForLongValue_theCorrectIndexOfTheValueIsReturned() throws Exception {
```
Comment (ashleyfrieze) [inline core-java-modules/core-java-reflection-2/src/test/java/com/baeldung/reflection/access/privatemethods/InvokePrivateMethodsUnitTest.java:15]: The method doesn't need to be `public`

**Review by ashleyfrieze — CHANGES_REQUESTED** [review]

File: core-java-modules/core-java-reflection-2/src/test/java/com/baeldung/reflection/access/privatemethods/InvokePrivateMethodsUnitTest.java (hunk 1 above)
Comment (bradhandy) [reply core-java-modules/core-java-reflection-2/src/test/java/com/baeldung/reflection/access/privatemethods/InvokePrivateMethodsUnitTest.java:10]: I made an assumption we were using JUnit 4 without checking the dependencies.  Oops.  I will keep this in mind for future updates.

File: core-java-modules/core-java-reflection-2/src/test/java/com/baeldung/reflection/access/privatemethods/InvokePrivateMethodsUnitTest.java (hunk 2 above)
Comment (bradhandy) [reply core-java-modules/core-java-reflection-2/src/test/java/com/baeldung/reflection/access/privatemethods/InvokePrivateMethodsUnitTest.java:15]: Same assumption here as I stated in reply to the previous comment.  👍

**Pull request merged**